From 703a5b7353fc34df4401bba226736176e1c62b7e Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Mon, 18 Apr 2011 12:26:51 -0700
Subject: [PATCH 1/2] Allow args changes to affect all kernel entries
 (#696971)

When passing --update-kernel=ALL the --args should apply to all kernel
entries, not just the first one.

Resolves: rhbz#696971
---
 grubby.c |    8 +-------
 1 files changed, 1 insertions(+), 7 deletions(-)

diff --git a/grubby.c b/grubby.c
index c984c04..7716069 100644
--- a/grubby.c
+++ b/grubby.c
@@ -1979,7 +1979,7 @@ int updateActualImage(struct grubConfig * cfg, const char * image,
     const char ** arg;
     int useKernelArgs, useRoot;
     int firstElement;
-    int *usedElements, *usedArgs;
+    int *usedElements;
     int doreplace;
 
     if (!image) return 0;
@@ -2014,9 +2014,6 @@ int updateActualImage(struct grubConfig * cfg, const char * image,
     useRoot = (getKeywordByType(LT_ROOT, cfg->cfi)
 	       && !multibootArgs);
 
-    for (k = 0, arg = newArgs; *arg; arg++, k++) ;
-    usedArgs = calloc(k, sizeof(*usedArgs));
-
     for (; (entry = findEntryByPath(cfg, image, prefix, &index)); index++) {
 
 	if (multibootArgs && !entry->multiboot)
@@ -2092,7 +2089,6 @@ int updateActualImage(struct grubConfig * cfg, const char * image,
         usedElements = calloc(line->numElements, sizeof(*usedElements));
 
 	for (k = 0, arg = newArgs; *arg; arg++, k++) {
-            if (usedArgs[k]) continue;
 
 	    doreplace = 1;
 	    for (i = firstElement; i < line->numElements; i++) {
@@ -2107,7 +2103,6 @@ int updateActualImage(struct grubConfig * cfg, const char * image,
                     continue;
 		if (!argMatch(line->elements[i].item, *arg)) {
                     usedElements[i]=1;
-                    usedArgs[k]=1;
 		    break;
                 }
             }
@@ -2177,7 +2172,6 @@ int updateActualImage(struct grubConfig * cfg, const char * image,
 	}
     }
 
-    free(usedArgs);
     free(newArgs);
     free(oldArgs);
 
-- 
1.7.7.6

